permissions/src/main/java/org/openmbee/sdvc/permissions/DefaultPermissionService.java

```diff
@@ -0,0 +1,737 @@
+                    }
+                    Optional<BranchGroupPerm> perm = branchGroupPermRepo.findByBranchAndGroupAndInheritedIsFalse(branch, group.get());
+                    perm.ifPresent(branchGroupPerm -> branchGroupPermRepo.delete(branchGroupPerm));
+                }
+                break;
+        }
+        recalculateInheritedPerms(branch);
+    }
+
+    @Override
+    public void setProjectInherit(boolean isInherit, String projectId) {
+        Optional<Project> project = projectRepo.findByProjectId(projectId);
+
+        if (!project.isPresent()) {
+            throw new PermissionException(HttpStatus.NOT_FOUND, "Project not found");
+        }
+
+        Project proj = project.get();
+        proj.setInherit(isInherit);
+        projectRepo.save(proj);
+    }
+
+    @Override
+    public void setBranchInherit(boolean isInherit, String projectId, String branchId) {
+        Optional<Branch> branch = branchRepo.findByProject_ProjectIdAndBranchId(projectId, branchId);
+
+        if (!branch.isPresent()) {
+            throw new PermissionException(HttpStatus.NOT_FOUND, "Branch not found");
+        }
+
+        Branch bran = branch.get();
+        bran.setInherit(isInherit);
+        branchRepo.save(bran);
+    }
+
+    @Override
+    public void setOrgPublic(boolean isPublic, String orgId) {
+        Optional<Organization> organization = orgRepo.findByOrganizationId(orgId);
+
+        if (!organization.isPresent()) {
+            throw new PermissionException(HttpStatus.NOT_FOUND, "Organization not found");
+        }
+
+        Organization org = organization.get();
+        org.setPublic(isPublic);
+        orgRepo.save(org);
+    }
+
+    @Override
+    public void setProjectPublic(boolean isPublic, String projectId) {
+        Optional<Project> project = projectRepo.findByProjectId(projectId);
+
+        if (!project.isPresent()) {
+            throw new PermissionException(HttpStatus.NOT_FOUND, "Project not found");
+        }
+
+        Project proj = project.get();
+        proj.setPublic(isPublic);
+        projectRepo.save(proj);
+    }
+
+    @Override
+    public boolean hasOrgPrivilege(String privilege, String user, String orgId) {
+        Optional<Privilege> priv = privRepo.findByName(privilege);
+        if (!priv.isPresent()) {
+            throw new PermissionException(HttpStatus.BAD_REQUEST, "No such privilege");
+        }
+
+        Optional<OrgUserPerm> orgUserPerm = orgUserPermRepo.findAllByOrganization_OrganizationIdAndUser_Username(orgId, user);
+        if (!orgUserPerm.isPresent()) {
+            throw new PermissionException(HttpStatus.NOT_FOUND, "Organization or Role not found");
+        }
+
+        Set<Privilege> privileges = orgUserPerm.get().getRole().getPrivileges();
+        if (privileges.contains(priv.get())) {
+            return true;
+        }
+
+        Optional<OrgUserPerm> orgGroupPerm = orgUserPermRepo.findAllByOrganization_OrganizationIdAndUser_Username(orgId, user);
```

Comment:
this should be using group perm repo
i would say have a collection<string> for groups passed in (would change the interface def), then findAllByOrganization_OrganizationIdAndGroup_NameIn to get the set of group perms